core/src/main/java/hudson/model/Executor.java

```diff
@@ -219,21 +224,7 @@ public void run() {
         try {
             if (induceDeath)        throw new ThreadDeath();
 
-            SubTask task;
-            // transition from idle to building.
-            // perform this state change as an atomic operation wrt other queue operations
-            synchronized (queue) {
-                workUnit.setExecutor(this);
-                queue.onStartExecuting(this);
-                if (LOGGER.isLoggable(FINE))
-                    LOGGER.log(FINE, getName()+" grabbed "+workUnit+" from queue");
-                task = workUnit.work;
-                executable = task.createExecutable();
-                workUnit.setExecutable(executable);
-            }
-            if (LOGGER.isLoggable(FINE))
-                LOGGER.log(FINE, getName()+" is going to execute "+executable);
-
+            SubTask task = setup1();
```

Comment:
-1 on this as it is likely to complicate merging with https://github.com/jenkinsci/jenkins/pull/1596.